workflow/controller/operator.go

```diff
@@ -4023,6 +4023,9 @@ func (woc *wfOperationCtx) setStoredWfSpec(ctx context.Context) error {
 		wfutil.JoinWorkflowMetaData(&woc.wf.ObjectMeta, &wfDefault.ObjectMeta)
 		workflowTemplateSpec = wftHolder.GetWorkflowSpec()
 	}
+	if len(woc.execWf.Spec.PodSpecPatch) > 0 && woc.controller.Config.WorkflowRestrictions.MustNotChangeSpec() {
```

Comment:
This also applies to Strict mode right? `MustUseReference` is more suitable perhaps

Reply:
I think templateReferencing: Secure has prohibited changes, maybe podSpecPatch can also exist in strict mode.